### PR TITLE
add metrics port

### LINKED ISSUE #1274: The K8s deployment file of S3 gateway should exports P8s metric port
**What would you like to be added**:

Export Prometheus metric port of S3 gateway in the [K8s deployment file](https://github.com/juicedata/juicefs/blob/main/deploy/juicefs-s3-gateway.yaml)

**Why is this needed**:

After export the Prometheus metric port, user could use [`ServiceMonitor`](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md) CRD provided by Prometheus Operator to collect metrics easily.

### PR DESCRIPTION
**Pull request by zwwhdls**
fix https://github.com/juicedata/juicefs/issues/1274

### CODE REVIEW

File: deploy/juicefs-s3-gateway.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
           command:
             - sh
             - -c
-            - juicefs gateway $(METAURL) $(NODE_IP):9000
```
Comment (davies) [inline deploy/juicefs-s3-gateway.yaml:32]: Why not use NODE_IP?

Comment (zwwhdls) [reply deploy/juicefs-s3-gateway.yaml:32]: Using pod ip, NODE_IP is env from podIP

```
          env:
            - name: NODE_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
```

Comment (davies) [reply deploy/juicefs-s3-gateway.yaml:32]: I meant `--metrics=${NODE_IP}:9567`

Comment (zwwhdls) [reply deploy/juicefs-s3-gateway.yaml:32]: Both ${NODE_IP} and 0.0.0.0 is ok, because it's separate net namespace in container

**Pull request merged**